crates/sui-indexer/src/indexer.rs

```diff
@@ -104,19 +105,14 @@ impl Indexer {
             store.persist_protocol_configs_and_feature_flags(chain_id)?;
         }
 
-        let cancel_clone = cancel.clone();
-        let (exit_sender, exit_receiver) = oneshot::channel();
-        // Spawn a task that links the cancellation token to the exit sender
-        spawn_monitored_task!(async move {
-            cancel_clone.cancelled().await;
-            let _ = exit_sender.send(());
-        });
-
+        let mut exit_senders = vec![];
+        let mut executors = vec![];
+        let progress_store = ShimIndexerProgressStore::new(vec![
+            ("primary".to_string(), primary_watermark),
+            ("object_snapshot".to_string(), object_snapshot_watermark),
+        ]);
         let mut executor = IndexerExecutor::new(
-            ShimIndexerProgressStore::new(vec![
-                ("primary".to_string(), primary_watermark),
-                ("object_snapshot".to_string(), object_snapshot_watermark),
-            ]),
+            progress_store.clone(),
```

Comment:
So both executors will have the same progress store which has two watermarks? Are we still going to have the same issue if there are two watermarks?

Reply:
good question! 
It's just a shim progress store with an underlying hashmap. It's mostly a no-op and is only used to initialize the system.
It's safe to have more entries than needed. 
The actual read from it and setup happens during the `executor.register` call